topology/probes/runc/runc.go

```diff
@@ -0,0 +1,415 @@
+)
+
+// Probe describes a Docker topology graph that enhance the graph
+type Probe struct {
+	common.RWMutex
+	*ns.Probe
+	state          int64
+	hostNs         netns.NsHandle
+	wg             sync.WaitGroup
+	watcher        *fsnotify.Watcher
+	paths          []string
+	containers     map[string]*container
+	containersLock common.RWMutex
+}
+
+type container struct {
+	namespace string
+	node      *graph.Node
+}
+
+func (probe *Probe) containerNamespace(pid int) string {
+	return fmt.Sprintf("/proc/%d/ns/net", pid)
+}
+
+type containerState struct {
+	path             string
+	ID               string `json:"id"`
+	InitProcessPid   int    `json:"init_process_pid"`
+	InitProcessStart int64  `json:"init_process_start"`
+	Config           struct {
+		Labels []string `json:"labels"`
+	} `json:"config"`
+}
+
+type createConfig struct {
+	Image   string
+	ImageID string
+	Labels  map[string]interface{}
+}
+
+func getLabels(raw []string) map[string]interface{} {
+	labels := make(map[string]interface{})
+
+	for _, label := range raw {
+		kv := strings.SplitN(label, "=", 2)
+		switch len(kv) {
+		case 1:
+			logging.GetLogger().Debugf("Label format should be key=value: %s", label)
+		case 2:
+			var value interface{}
+			if err := json.Unmarshal([]byte(kv[1]), &value); err != nil {
+				common.SetField(labels, kv[0], common.NormalizeValue(kv[1]))
+			} else {
+				common.SetField(labels, kv[0], common.NormalizeValue(value))
+			}
+		}
+	}
+
+	return labels
+}
+
+func getCreateConfig(path string) (*createConfig, error) {
+	body, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to read create config %s: %s", path, err)
+	}
+
+	var cc createConfig
+	if err := json.Unmarshal(body, &cc); err != nil {
+		return nil, fmt.Errorf("Unable to parse create config %s: %s", path, err)
+	}
+
+	return &cc, nil
+}
+
+func getStatus(state *containerState) string {
+	if state.InitProcessPid == 0 {
+		return "stopped"
+	}
+
+	info, err := common.GetProcessInfo(state.InitProcessPid)
+	if err != nil {
+		return "stopped"
+	}
+	if info.Start != state.InitProcessStart || info.State == common.Zombie || info.State == common.Dead {
+		return "stopped"
+	}
+	base := path.Base(state.path)
+	if _, err := os.Stat(path.Join(base, "exec.fifo")); err == nil {
+		return "created"
+	}
+	return "running"
+}
+
+func parseState(path string) (*containerState, error) {
+	body, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to read container state %s: %s", path, err)
+	}
+
+	var state containerState
+	if err := json.Unmarshal(body, &state); err != nil {
+		return nil, fmt.Errorf("Unable to parse container state %s: %s", path, err)
+	}
+
+	state.path = path
+
+	return &state, nil
+}
+
+func getMetadata(state *containerState) graph.Metadata {
+	m := graph.Metadata{
+		"ContainerID":  state.ID,
+		"ContainerPID": int64(state.InitProcessPid),
+		"Status":       getStatus(state),
+	}
+
+	labels := getLabels(state.Config.Labels)
```

Comment:
oh, you missed a oneliner !

Reply:
no I wanted to see if you were still able to detect it, I am reassured :)